### PR TITLE
Improve functions

### LINKED ISSUE #9: [Code Quality] Improve calculateResult() so the name of function is fully truthy
It's not explicit by reading its name that calculateResult() already shows the result of calculating the expression in the calculator's screen. 
The call to showOnScreen() should be left outside the function OR calculateResult() name should be corrected to something like calculateResultAndShowOnScreen().

### PR DESCRIPTION
**Pull request by jesslourenco**
- Cleaned calculateResult() so it really just calculated the result
- Noticed a lot of code duplication with minimal changes after getting the result. So created an additional function doAfterOperation() that checks for long decimal numbers and run code depending on type of operation being done
- Improved showOnScreen() so the  algo can call it if I want to append numbers or replace it on the screen
- While doing these, found a bug regarding operator signs:
     - click 5
     - click plus sign
     - click minus sign
     - expected outcome: the minus sign replaces the plus sign in the expression
     - actual outcome: calculator perform weird operation
It was an easy fix, so I fixed it already on this branch.

### CODE REVIEW

File: calculator.js [hunk 1]
```diff
@@ -115,24 +116,22 @@ function handlesNumber(value){
...
-    showOnScreen(value);
+    showOnScreen(value,'add');
 }
 
 function handlesOperator(value){
 
     if (!operator){
         num1 = digitsOnScreen.innerHTML;
-    }else if(num1) {
+    }else if(num1 && !operatorState) {
```
Comment (shankarnakai) [inline calculator.js:126]: `operatorState` is means that a signal was clicked right?
In the scenario:
- Click number
- Click plus 
- Click number

Will this continue to work?

File: calculator.js [hunk 2]
```diff
@@ -157,18 +156,20 @@ function handlesEquals(){
...
 
 function handlesDecSeparator(value){
-    showOnScreen(value);
+    showOnScreen(value,'add');
     btnDecSeparator.className += ' dot-disabled';
 }
 
-function showOnScreen(value){
-    digitsOnScreen.innerHTML += value;
+function showOnScreen(value, operation='replace'){
```
Comment (shankarnakai) [inline calculator.js]: It's preferable to build 2 distinct function to maintain it simple.
- putOnScreen - can be your replace operation
- appendOnScreen - can be your add operation

Other thing is the operator is a string, so if we want to change in the future this labels `add` and `replace` the developer responsible for this change will need to chase all the places in the code to change, open opportunities to mistakes.

Using functions we can use some editor to replace the function name.
Functions with a single responsibility reveal better the intention that they are doing.

File: calculator.js [hunk 3]
```diff
@@ -191,16 +192,37 @@ function calculateResult(operator, num1, num2){
...
-        result = result.toFixed(9);
-    }
-    digitsOnScreen.innerHTML = result;
-    console.log(result);
+    }    
+
     return result;
 }
 
+function doAfterGettingResult(result, opType){
```
Comment (shankarnakai) [inline calculator.js]: The name of the function is not really saying what it is doing, but stating that the developer want to use this function after get the result.

The same suggestion made in the `showOnScreen` is applicable here.

Notice that the `showOnScreen` is not passing anything, making this call inconsistent with the other calls in the code.

**Review by shankarnakai — CHANGES_REQUESTED** [review]
I would like to verify if we can get a better solution for the `doAfterGetResult`. 
I'm wondering if we can have a function reset that will return the calculator to an initial state.

File: calculator.js (hunk 1 above)
Comment (jesslourenco) [reply calculator.js:126]: Happy path still works, QA tested

File: calculator.js (hunk 2 above)
Comment (jesslourenco) [reply calculator.js]: ok, broke the function into replaceOnScreen() and addToScreen()

File: calculator.js (hunk 3 above)
Comment (jesslourenco) [reply calculator.js]: got rid of the function after realizing that a good chunk of the code could be (and should be) performed by handlesReset()

**Review by shankarnakai — APPROVED** [review]

**Pull request merged**